libflux/flux-core/src/semantic/tests/labels.rs

```diff
@@ -0,0 +1,226 @@
+              │                                            ^^^^^^
+
+            error: string is not a label
+              ┌─ main:3:31
+              │
+            3 │             x = [{ a: 1 }] |> fill(column: column, value: "x")
+              │                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+        "#]],
+    }
+}
+
+#[test]
+fn undefined_field() {
+    test_error_msg! {
+        config: AnalyzerConfig{
+            features: vec![Feature::LabelPolymorphism],
+            ..AnalyzerConfig::default()
+        },
+        env: map![
+            "fill" => "(<-tables: [{ A with B: C }], ?column: B, ?value: D) => [{ A with B: D }]
+                where B: Label
+                "
+        ],
+        src: r#"
+            x = [{ b: 1 }] |> fill(column: "a", value: "x")
+        "#,
+        expect: expect![[r#"
+            error: record is missing label a
+              ┌─ main:2:31
+              │
+            2 │             x = [{ b: 1 }] |> fill(column: "a", value: "x")
+              │                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+        "#]],
+    }
+}
+
+#[test]
+fn merge_labels_to_string() {
+    test_infer! {
+        config: AnalyzerConfig{
+            features: vec![Feature::LabelPolymorphism],
+            ..AnalyzerConfig::default()
+        },
+        src: r#"
+            x = if 1 == 1 then "a" else "b"
+            y = if 1 == 1 then "a" else "b" + "b"
+            z = ["a", "b"]
+        "#,
+        exp: map![
+            "x" => "string",
+            "y" => "string",
+            "z" => "[string]",
+        ],
+    }
+}
+
+#[test]
+fn merge_labels_to_string_in_function() {
+    test_infer! {
+        config: AnalyzerConfig{
+            features: vec![Feature::LabelPolymorphism],
+            ..AnalyzerConfig::default()
+        },
+        env: map![
+            "same" => "(x: A, y: A) => A"
+        ],
+        src: r#"
+            x = same(x: "a", y: "b")
+            y = same(x: ["a"], y: ["b"])
+        "#,
+        exp: map![
+            "x" => "string",
+            "y" => "[string]",
+        ],
+    }
+}
+
+#[test]
+fn attempt_to_use_label_polymorphism_without_feature() {
+    test_error_msg! {
+        env: map![
+            "columns" => "(table: A, ?column: C) => { C: string } where A: Record, C: Label",
+        ],
+        src: r#"
+            x = columns(table: { a: 1, b: "b" }, column: "abc")
+            y = x.abc
+        "#,
+        expect: expect![[r#"
+            error: string is not Label (argument column)
+              ┌─ main:2:58
+              │
+            2 │             x = columns(table: { a: 1, b: "b" }, column: "abc")
+              │                                                          ^^^^^
+
+            error: record is missing label abc
+              ┌─ main:3:17
+              │
+            3 │             y = x.abc
+              │                 ^
+
+        "#]],
+    }
+}
```

Comment:
To make sure I understand this, we can define a builtin type that requires a label, but if we don't enable the feature you won't be able to use that builtin value? So this test case shows that the label inference logic is disabled.

Reply:
Yes. More precisely that we aren't creating any label types without the feature (which in turn prevents the builtin from being used).